algorithms/gr-dfsrh/javascript/dfs.js

```diff
@@ -0,0 +1,43 @@
+
+const dfs = (graph, startNode) => {
+    // Create an array / fake "stack" for storing next visiting nodes
+    let stack = [startNode];
+    let visited = [];
+
+    while (stack.length) {
+        // Take first node from stack to visit
+        let node = stack.shift();
+
+        // Check whether the node is not visited 
+        if (visited.indexOf(node) === -1) {
+            visited.push(node);
+            /* eslint-disable no-console */
+            console.log(`visited ${node}`);
+            /* eslint-enable no-console */
+            
+            // Find direct-connected node and put to stack if it's not visited
+            graph.edges.filter(edge => edge.start === node && !visited.includes(edge.end)).forEach((item) => {
```

Comment:
Codacy found an issue: [Function declared in a loop contains unsafe references to variable(s) 'stack', 'stack'.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396906)